lightning/src/ln/channelmanager.rs

```diff
@@ -3996,7 +3996,31 @@ where
 		self.pending_outbound_payments.test_set_payment_metadata(payment_id, new_payment_metadata);
 	}
 
-	pub(super) fn send_payment_for_bolt12_invoice(&self, invoice: &Bolt12Invoice, payment_id: PaymentId) -> Result<(), Bolt12PaymentError> {
+	/// Pays the [`Bolt12Invoice`] associated with the `payment_id` encoded in its `payer_metadata`.
+	///
+	/// The invoice's `payer_metadata` is used to authenticate that the invoice was indeed requested
+	/// before attempting a payment. [`Bolt12PaymentError::UnexpectedInvoice`] is returned if this
+	/// fails or if the encoded `payment_id` is not recognized. The latter may happen once the
+	/// payment is no longer tracked because the payment was attempted after:
+	/// - an invoice for the `payment_id` was already paid,
+	/// - one timer tick has elapsed since initially requesting the invoice when paying an offer, or
```

Comment:
Timer ticks for `ChannelManager` should be one minute. And this may actually be up to two timer ticks as implemented. So really at least one *full* timer tick.

https://github.com/lightningdevkit/rust-lightning/blob/b8cdde8af64189fd1567cbc85cfae26c610af969/lightning/src/ln/outbound_payment.rs#L1596-L1608